Mastermind/Mastermind.cs

```diff
@@ -1,86 +1,145 @@
 ﻿using System;
+using System.Collections.Generic;
 
-namespace Mastermind
-{
-    class Program
-    {
-        // possible letters in code
-        public static char[] letters = new char[] { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h' };
-        
-        // size of code
+namespace Mastermind {
+    class Program {
+        //added code to randomly generate solution
+        public static string[] letters = new string[] { "a", "b", "c", "d", "e", "f"};
         public static int codeSize = 4;
-        
-        // number of allowed attempts to crack the code
-        public static int allowedAttempts = 10;
-        
-        // number of tried guesses
-        public static int numTry = 0;
-        
-        // test solution
-        public static char[] solution = new char[] {'a', 'b', 'c', 'd'};
-        
-        // game board
-        public static string[][] board = new string[allowedAttempts][];
-        
-        
-        public static void Main()
-        {
-            char[] guess = new char[4];
-
-            CreateBoard();
-            DrawBoard();
-            Console.WriteLine("Enter Guess:");
-            guess = Console.ReadLine().ToCharArray();
+        public static string[] solution = new string[] {"a", "b", "c", "d"};
 
-            // leave this command at the end so your program does not close automatically
-            Console.ReadLine();
-        }
-        
-        public static bool CheckSolution(char[] guess)
-        {
-            // Your code here
-
-            return false;
-        }
-        
-        public static string GenerateHint(char[] guess)
-        {
-            // Your code here
-            return " ";
-        }
-        
-        public static void InsertCode(char[] guess)
+        public static void GenerateRandomCode() 
         {
-            // Your code here
+            Random rnd = new Random();
+            for(var i = 0; i < codeSize; i++)
+            {
+                solution[i] = letters[rnd.Next(0, letters.Length)];
+            }
         }
-        
-        public static void CreateBoard()
-        {
-            for (var i = 0; i < allowedAttempts; i++)
+        static void Main (string[] args) {
+            //added code to pass maximum number of tries into game
+            int maxTries = 10;
+            //generates random solution; can comment this out to run with test solution "abcd"
+            GenerateRandomCode();
+            Game game = new Game (maxTries, solution);
```

Comment:
I dont see the Game class making use of maxTries to manage how many guesses are left